cleancode/src/main/java/com/epam/cleancode/comments/MortgageInstallmentCalculator.java

```diff
@@ -1,39 +1,31 @@
 package com.epam.cleancode.comments;
 
-public class MortgageInstallmentCalculator {
+class MortgageInstallmentCalculator {
+
+    private static final String INVALID_INPUT_EXCEPTION = "Negative values are not allowed";
+
+    MortgageInstallmentCalculator() {
+    }
 
     /**
-     *
-     * @param p principal amount
-     * @param t term of mortgage in years
-     * @param r rate of interest
+     * @param principalAmount       principal amount
+     * @param termOfMortgageInYears term of mortgage in years
+     * @param rateOfInterest        rate of interest
      * @return monthly payment amount
      */
-    public static double calculateMonthlyPayment(
-            int p, int t, double r) {
-
-        //cannot have negative loanAmount, term duration and rate of interest
-        if (p < 0 || t <= 0 || r < 0) {
-            throw new InvalidInputException("Negative values are not allowed");
+    static double calculateMonthlyPayment(
+            int principalAmount, int termOfMortgageInYears, double rateOfInterest) {
+        double termInMonths = termOfMortgageInYears * 12.0;
+        double rateInDecimal = rateOfInterest / 100.0;
+        double monthlyRate = rateInDecimal / 12.0;
+
+        if (principalAmount < 0 || termOfMortgageInYears <= 0 || rateOfInterest < 0) {
```

Comment:
Can be extracted to a separate method, where you can validate function parameters